### PR TITLE
patchkernel: add the concept of "adaption mode"

### PR DESCRIPTION
**Pull request by andrea-iob**
Adaption mode tells if the patch can be adapted and which strategies can be used to adapt the patch.

The following adaption modes are supported:
 - disabled, no adaption can be performed;
 - automatic, adaption is performed specifying which cells should be refined/coarsen and then the patch will perform all the alterations needed to fulfill the adaption requests;
 - manual, this modes allows to use low level function to add and delete individual cells and vertices. It's up to the user to guarantee the consistency of the patch.

Adaption mode replaces the concept of "expert mode". The functions "isExpert" and "setExpert" are now deprecated.

Depends on #370.

### CODE REVIEW

File: src/lineunstructured/lineunstructured.cpp [hunk 1]
```diff
@@ -74,15 +74,15 @@ LineUnstructured::LineUnstructured()
     among the processes
 */
 LineUnstructured::LineUnstructured(int dimension, MPI_Comm communicator)
-    : LineKernel(PatchManager::AUTOMATIC_ID, dimension, communicator, 1, true)
+    : LineKernel(PatchManager::AUTOMATIC_ID, dimension, communicator, 1, ADAPTION_MANUAL, PARTITIONING_DISABLED)
```
Comment (edoardolombardi) [inline src/lineunstructured/lineunstructured.cpp:77]: Why PARTITIONING_DISABLED?

Comment (andrea-iob) [reply src/lineunstructured/lineunstructured.cpp]: Fixed.

**Comment by andrea-iob** [comment]
I've removed all the changes not directly related to this pull request.

**Review by marcocisternino — APPROVED** [review]
check code and tested in application using voloctree.
Tests using manual adapted volunstructured or surfunstructured would test the branch better than I did.

**Review by edoardolombardi — APPROVED** [review]

**Pull request merged**